### PR TITLE
ADP-119 Add RTS and EOI tables to Partner EOI Page

### PR DESCRIPTION
**Pull request by nick-mcguffin**
---
name: ADP-119 Add RTS and EOI tables to Partner EOI Page
about: Request to merge code
title: 'ADP-119 Add RTS and EOI tables to Partner EOI Page'

---

**PR Type** *(Select all that apply)*
  - [x] Feature 
  - [ ] Bug fix
  - [x] Improvement 
  - [ ] Other

**Feature Description**

Add Expression Of Interest and Request To Supply table UI to Partner Expressions of Interest Page.

**Submission Checklist** *(do not remove)*

Use the following as a guide to ensure your PR is complete and acceptable:

- [x] You are NOT attempting to merge into the `master` branch
- [x] You are requesting to merge your branch into the `development` branch or one of its sub-branches
- [x] Your code compiles and runs without errors (on as many operating systems as practicable
- [x] Code has sufficient Javadoc comments (giving examples where needed)
- [x] Code is clear and readable, being sure to use meaningful names and effective whitespace
- [x] Appropriate [naming convention](https://en.wikipedia.org/wiki/Naming_convention_(programming)) is used according to the language used
- [x] Code duplication is minimized through use of polymorphism and inheritance
- [x] Dependencies have been checked for vulnerabilities
- [x] There is sufficient unit tests for the code and all tests pass
- [x] Unit tests utilize mock data/tools and don't write data to any databases
- [x] There is no logic in any views, use service/controller methods as helpers 
- [x] Use spaces between method parameters `[1, 2, 3]` not `[1,2,3]` and around operators `a + b` not `a+b`
- [x] All work has been thoroughly checked for grammatical and spelling errors
- [x] Documentation is to be clear, concise, and written in English
- [x] Changes are documented in a `README.md` if applicable

### CODE REVIEW

**Comment by nick-mcguffin** [comment]
Similar to the Educator EOI page. 

First table shows open EOI's made by Educators. Partners will be able to apply to these open EOI's.

Second table shows open Requests to Supply Work, made by Partners. Partners will be able to add another Request to Supply

Third table shows a history of an educators Requests to Supply Work. No actions required on these - just a visual history.

File: web/src/main/resources/templates/fragments/expressions-of-interest-tables.html [hunk 1]
```diff
@@ -170,45 +170,100 @@ <h1 class="table-heading">Accepted Requests To Supply Services</h1>
...
+    <div class="row table-container">
+        <table class="table table-sm table-primary table-striped table-hover">
+            <thead>
+                <tr>
+                    <th>ID</th>
+                    <th>Category</th>
+                    <th>Location</th>
+                    <th>Description</th>
+                    <th>Date Opened</th>
+                    <th class="text-center">Manage</th>
```
Comment (nathansnow1981) [inline web/src/main/resources/templates/fragments/expressions-of-interest-tables.html:191]: I don't think educators need access to edit or delete an expression of interest
```suggestion
<th class="text-center"></th>
```

File: web/src/main/resources/templates/fragments/expressions-of-interest-tables.html [hunk 2]
```diff
@@ -170,45 +170,100 @@ <h1 class="table-heading">Accepted Requests To Supply Services</h1>
...
+                    <td th:text="${openEOI.location}">Brisbane</td>
+                    <td th:text="${openEOI.description}">Slavey with extra steps</td>
+                    <td th:text="${#dates.format(openEOI.date, 'dd/MM/yyyy')}">01/01/2022</td>
+                    <td style="display: flex;">
+                        <div class="mx-auto">
+                            <button class="form-button">Apply</button>
+                            <span class="material-symbols-rounded btn-edit" title="Edit">edit</span>
+                            <span class="material-symbols-rounded btn-delete">delete</span>
+                        </div>
+                    </td>
```
Comment (nathansnow1981) [inline web/src/main/resources/templates/fragments/expressions-of-interest-tables.html:207]:
```suggestion
<td style="display: flex;">
    <div class="mx-auto">
        <button class="form-button">Apply</button>
    </div>
</td>
```

**Review by nathansnow1981 — APPROVED** [review]
Nice 👌 I think this layout is easier to follow
I'm beginning to wonder if the whole RTS thing is substantial enough to deserve it's own page and menu item?
It may be a candidate for the bottom of the backlog 🤷‍♀️

**Review by nathansnow1981 — APPROVED** [review]

**Pull request merged**